// @electron/remote/index.js
import { MessageBox } from 'element-ui'
import { notSupport } from 'utils'

const MESSAGE_TYPE_MAP = new Map([
  ['info', 'info'],
  ['error', 'error'],
  ['warning', 'warning'],
  ['question', 'info']
])

export const app = {
  getVersion () {
    return 'motrix-web'
  }
}

export const shell = {}

export const nativeTheme = {}

export const getCurrentWindow = () => {
  return {
    minimize () {
      notSupport()
    },

    maximize () {
      notSupport()
    },

    unmaximize () {
      notSupport()
    },

    isMaximized () {
      notSupport()
    },

    close () {
      notSupport()
    }
  }
}

export const webContents = {}

export const dialog = {
  showMessageBox (options) {
    return MessageBox({
      type: MESSAGE_TYPE_MAP.get(options.type),
      message: options.message,
      title: options.title,
      confirmButtonText: options.buttons[0],
      cancelButtonText: options.buttons[1],
      showCancelButton: true
    })
      .then(() => {
        return Promise.resolve({
          response: 0,
          checkboxChecked: false
        })
      })
      .catch((e) => {
        return Promise.resolve({
          response: 1,
          checkboxChecked: false
        })
      })
  },

  showOpenDialog () {
    notSupport()
    return new Promise(resolve => {
      resolve({ canceled: true, filePaths: [] })
    })
  }
}
